server/api/routes/index.js

```diff
@@ -1,26 +1,28 @@
 import express from 'express';
-import Users from '../controllers/userController';
-import Trip from '../controllers/tripController';
-import Booking from '../controllers/bookingController';
+import UserController from '../controllers/userController';
+import TripController from '../controllers/tripController';
+import BookingController from '../controllers/bookingController';
 import Auth from '../middleware/auth';
 import allValidations from '../middleware/validations';
 
+
 const router = express.Router();
   
 //authentication routes
-router.post('/auth/signup', allValidations.signup, allValidations.validateEmail, Users.signup);
-router.post('/auth/signin', allValidations.signin, Users.signin);
+router.post('/auth/signup', allValidations.signup, allValidations.validateEmail, UserController.signup);
+router.post('/auth/signin', allValidations.signin, UserController.signin);
 
 //trip routes
-router.post('/trips', Auth.checkAdmin, allValidations.validateTrip, Trip.createNewTrip);
-router.get('/trips', Trip.getAllTrips);
-router.get('/trips/:id',  allValidations.validateId, Trip.getSpecificTrip);
-router.patch('/trips/:id/cancel', Auth.checkAdmin, allValidations.validateId, Trip.cancelTrip);
+router.post('/trips', Auth.checkAdmin, allValidations.validateTrip, TripController.createNewTrip);
+router.get('/trips', TripController.getAllTrips);
+router.get('/trips/:id',  allValidations.validateId, TripController.getSpecificTrip);
+router.patch('/trips/:id/cancel', Auth.checkAdmin, allValidations.validateId, TripController.cancelTrip);
+router.get('/trip/:origin', allValidations.validateOrigin, TripController.getAllTripsByOrigin);
```

Comment:
@nziokaivy  You don't need another endpoint for filtering, just use the same as to get all trips but use request's queries to provide filter data for instance: `/trips?origin=Kigali`, `/trips?origin=Kigali&destination=Tanzania`